syntax = "proto3";

package myrtlespeech.protos;

import "myrtlespeech/protos/ctc_beam_decoder.proto";
import "myrtlespeech/protos/ctc_greedy_decoder.proto";
import "myrtlespeech/protos/ctc_loss.proto";
import "myrtlespeech/protos/deep_speech_1.proto";
import "myrtlespeech/protos/deep_speech_2.proto";
import "myrtlespeech/protos/pre_process_step.proto";
import "myrtlespeech/protos/rnn_t.proto";
import "myrtlespeech/protos/rnn_t_loss.proto";
import "myrtlespeech/protos/rnn_t_greedy_decoder.proto";
import "myrtlespeech/protos/rnn_t_beam_decoder.proto";

message SpeechToText {
  // Output alphabet.
  string alphabet = 1;

  repeated PreProcessStep pre_process_step = 2;

  // Supported networks.
  oneof supported_models {
    DeepSpeech1 deep_speech_1 = 3;
    DeepSpeech2 deep_speech_2 = 4;
    RNNT rnn_t = 8;
  }

  // Supported loss functions.
  oneof supported_losses {
    CTCLoss ctc_loss = 5;
    RNNTLoss rnn_t_loss = 9;
  }

  // Supported post processing of network output.
  oneof supported_post_processes {
    CTCGreedyDecoder ctc_greedy_decoder = 6;
    CTCBeamDecoder ctc_beam_decoder = 7;
    RNNTGreedyDecoder rnn_t_greedy_decoder = 10;
    RNNTBeamDecoder rnn_t_beam_decoder = 11;

  }
}
